presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithExchangeMaterialization.java

```diff
@@ -380,5 +380,25 @@ protected boolean supportsNotNullColumns()
         return false;
     }
 
+    @Test
+    public void testEmptyBucketedTemporaryTable()
+    {
+        assertQuery("SELECT COUNT(DISTINCT linenumber), COUNT(*) from lineitem where linenumber < 0");
+    }
+
+    @Test
+    public void testBucketedTemporaryTableWithMissingFiles()
+    {
+        testBucketedTemporaryTableWithMissingFiles(true);
+        testBucketedTemporaryTableWithMissingFiles(false);
+    }
+
+    private void testBucketedTemporaryTableWithMissingFiles(boolean isFileRenameEnabled)
+    {
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "file_renaming_enabled", String.valueOf(isFileRenameEnabled))
```

Comment:
`file_renaming_enabled` will be effective when Presto is creating files for an INSERT statement. Ideal test would be to do an INSERT when `create-empty-bucket-files-for-temporary-table =false` and `file_renaming_enabled=true`.

Reply:
This query will generate temporary table with new filesnamed "1", for example, I could get `invalid hive bucket file name: 1` error if I run this test but does not support numerical name.